libs/gen/EnumsSourceGenerator.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System.CodeDom.Compiler;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Garnet;
+
+[Generator]
+public class EnumsSourceGenerator : IIncrementalGenerator
+{
+    const string GeneratedClassName = "EnumUtils";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var enumDetails = context.SyntaxProvider
+            .ForAttributeWithMetadataName(
+                "Garnet.common.GenerateEnumDescriptionUtilsAttribute",
+                predicate: static (_, _) => true,
```

Comment:
```suggestion
                predicate: static (node, _) => node is EnumDeclarationSyntax,
```

fast check to make the pipeline little bit more rigorous :)

Reply:
I'm not sure about this, `GenerateEnumDescriptionUtilsAttribute` is already limited to be used only on enums

Reply:
Sure, that works too.